cmd/argocd/commands/app.go

```diff
@@ -2065,38 +2065,153 @@ func printTreeViewDetailed(nodeMapping map[string]argoappv1.ResourceNode, parent
 	_ = w.Flush()
 }
 
+type SyncOptionsOverrideStyle string
+
+var (
+	SyncOptionsOverrideReplace SyncOptionsOverrideStyle = "replace"
+	SyncOptionsOverridePatch   SyncOptionsOverrideStyle = "patch"
+)
+
+func SyncOptionsOverrideStyleFromString(s string) (SyncOptionsOverrideStyle, error) {
+	switch s {
+	case string(SyncOptionsOverrideReplace):
+		return SyncOptionsOverrideReplace, nil
+	case string(SyncOptionsOverridePatch):
+		return SyncOptionsOverridePatch, nil
+	}
+	return "", fmt.Errorf("unknown sync options override style : %s", s)
+}
+
+const (
+	syncOptionKeyReplace            = "Replace"
+	syncOptionKeyServerSideApply    = "ServerSideApply"
+	syncOptionKeyApplyOutOfSyncOnly = "ApplyOutOfSyncOnly"
+	cliFlagKeyReplace               = "replace"
+	cliFlagKeyServerSideApply       = "server-side"
+	cliFlagKeyApplyOutOfSyncOnly    = "apply-out-of-sync-only"
+)
+
+// Known sync option keys supported from CLI
+var knownSyncOptionKeys = []string{syncOptionKeyReplace, syncOptionKeyServerSideApply, syncOptionKeyApplyOutOfSyncOnly}
+
+// parseKnownMap converts a []string of form Key=true|false into a map for the known keys only
+func parseKnownMap(items []string) map[string]bool {
+	m := map[string]bool{}
+	for _, it := range items {
+		parts := strings.SplitN(it, "=", 2)
+		key := parts[0]
+		switch key {
+		case syncOptionKeyReplace, syncOptionKeyServerSideApply, syncOptionKeyApplyOutOfSyncOnly:
+			m[key] = parts[1] == "true"
+		}
+	}
+	return m
+}
+
+// serializeKnownMap converts known map back to []string in a stable key order
+func serializeKnownMap(m map[string]bool) []string {
+	out := make([]string, 0, len(m))
+	for _, k := range knownSyncOptionKeys {
+		if v, ok := m[k]; ok {
+			out = append(out, fmt.Sprintf("%s=%t", k, v))
+		}
+	}
+	return out
+}
+
+// cliMapFromFlags builds a map from flags that were explicitly changed
+func cliMapFromFlags(cmd *cobra.Command, replace, serverSideApply, applyOutOfSyncOnly bool) (map[string]bool, bool) {
+	changed := cmd.Flags().Changed(cliFlagKeyReplace) || cmd.Flags().Changed(cliFlagKeyServerSideApply) || cmd.Flags().Changed(cliFlagKeyApplyOutOfSyncOnly)
+	m := map[string]bool{}
+	if cmd.Flags().Changed(cliFlagKeyReplace) {
+		m[syncOptionKeyReplace] = replace
+	}
+	if cmd.Flags().Changed(cliFlagKeyServerSideApply) {
+		m[syncOptionKeyServerSideApply] = serverSideApply
+	}
+	if cmd.Flags().Changed(cliFlagKeyApplyOutOfSyncOnly) {
+		m[syncOptionKeyApplyOutOfSyncOnly] = applyOutOfSyncOnly
+	}
+	return m, changed
```

Comment:
```suggestion
	return m, changed
```
I see why are you returning changed here. However, we can simplify this by simply returning `len(m) > 0` i.e `return m, len(m) > 0`